desktop/main/sources/currentNetwork.ts

```diff
@@ -1,14 +1,46 @@
 import { find } from 'ramda';
-import { combineLatest, map, Subject } from 'rxjs';
+import {
+  combineLatestWith,
+  distinctUntilChanged,
+  from,
+  iif,
+  map,
+  mergeWith,
+  Observable,
+  Subject,
+  switchMap,
+} from 'rxjs';
 import { Network, Wallet } from '../../../shared/types';
+import NodeConfig from '../NodeConfig';
 
 export default (
+  $runNodeBeforeLogin: Observable<boolean>,
   $wallet: Subject<Wallet | null>,
   $networks: Subject<Network[]>
 ) =>
-  combineLatest([$wallet, $networks]).pipe(
+  $networks.pipe(
+    combineLatestWith($runNodeBeforeLogin),
+    mergeWith($wallet),
+    distinctUntilChanged(),
+    switchMap((input) => {
+      // @ts-ignore
```

Comment:
I removed the ts-ignore, but still have to force definition for the needed attribute from the object, because boolean defenatly doesn't have this attribute 
<img width="438" alt="Screenshot 2022-08-14 at 08 12 08" src="https://user-images.githubusercontent.com/54288612/184524868-b9a32171-a145-4087-ac42-a187e58259c9.png">